### ISSUE TITLE
error after save update content if use Falang

### ISSUE THREAD

**Issue by lungkao** [FLEXIcontent/flexicontent-cck#964]
An error has occurred.
1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ') ) AND i.language = '*' AND i.id <> 3' at line 1

and  have
Warning: trim() expects parameter 1 to be string, array given in /Applications/MAMP/htdocs/joomla39/components/com_flexicontent/models/category.php on line 1212

if use FaLang Language Switcher

Joomla! 3.9.13  Flexicontent 3.3.1.6 falang 3.2.0

**Comment by ggppdk** [comment]
I ll test
Can you print the full SQL error ?

**Comment by lungkao** [comment]
> I ll test
> Can you print the full SQL error ?

no  error in  log  file

**Comment by ggppdk** [comment]
1. Enable Joomla debug and you will get a trace that shows which line created the SQL query
2. Then to be able to print the full SQL statement
Find inside your template's error.php
```
<?php if ($this->debug) : ?>
	<br/><?php echo htmlspecialchars($this->error->getFile(), ENT_QUOTES, 'UTF-8');?>:<?php echo $this->error->getLine(); ?>
```
and after the above add (temporarily and then remove after you copy the SQL error)
```
	<?php if ($this->error instanceof JDatabaseExceptionExecuting) : ?>
		<br/><?php echo htmlspecialchars($this->error->getQuery(), ENT_QUOTES, 'UTF-8');?>
	<?php endif; ?>

```

**Comment by lungkao** [comment]
1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ') ) AND i.language = '*' AND i.id <> 1' at line 1
/Applications/MAMP/htdocs/joomla39/libraries/joomla/database/driver/mysqli.php:665
SELECT COUNT(*) FROM aid2j_flexicontent_items_tmp AS i JOIN aid2j_flexicontent_items_ext AS e ON i.id = e.item_id LEFT JOIN aid2j_flexicontent_cats_item_relations AS rel ON i.id = rel.itemid WHERE i.alias='2019-12-17-13-12-11-th-th' AND (i.catid=0 OR rel.catid IN () ) AND i.language = '*' AND i.id <> 1

**Comment by lungkao** [comment]
![Screen Shot 2562-12-22 at 23 56 46](https://user-images.githubusercontent.com/2047778/71324849-c00ff180-2516-11ea-8882-697f7dae975c.png)

**Comment by ggppdk** [comment]
So this occurs in new item form , backend and frontend ?

**Comment by lungkao** [comment]
> So this occurs in new item form , backend and frontend ?
Both

**Comment by ggppdk** [comment]
Ok i think this is an easy fix, i will test and fix tomorrow morning,
and ask you to test the fix too

**Comment by ggppdk** [comment]
Fixed with 998635d41

**Comment by lungkao** [comment]
> Fixed with [998635d](https://github.com/FLEXIcontent/flexicontent-cck/commit/998635d419b8b422a88d0426c0149fbcdf0eed7a)

Test pass .  Thank  you  for hardwork
